api/routes.go

```diff
@@ -24,75 +24,84 @@ var (
-	api.staticRouter.DELETE("/user/uploads/:skylink", api.WithDBSession(api.withAuth(api.userUploadsDELETE)))
-	api.staticRouter.GET("/user/downloads", api.WithDBSession(api.withAuth(api.userDownloadsGET)))
+	api.staticRouter.GET("/user/stats", api.mustAuth(api.userStatsGET))
+	api.staticRouter.GET("/user/pubkey/register", api.WithDBSession(api.mustAuth(api.userPubKeyRegisterGET)))
+	api.staticRouter.POST("/user/pubkey/register", api.WithDBSession(api.mustAuth(api.userPubKeyRegisterPOST)))
+	api.staticRouter.GET("/user/uploads", api.WithDBSession(api.mustAuth(api.userUploadsGET)))
+	api.staticRouter.DELETE("/user/uploads/:skylink", api.WithDBSession(api.mustAuth(api.userUploadsDELETE)))
+	api.staticRouter.GET("/user/downloads", api.WithDBSession(api.mustAuth(api.userDownloadsGET)))
 
 	// Endpoints for user API keys.
-	api.staticRouter.POST("/user/apikeys", api.WithDBSession(api.withAuth(api.userAPIKeyPOST)))
-	api.staticRouter.GET("/user/apikeys", api.WithDBSession(api.withAuth(api.userAPIKeyGET)))
-	api.staticRouter.DELETE("/user/apikeys/:id", api.WithDBSession(api.withAuth(api.userAPIKeyDELETE)))
+	api.staticRouter.POST("/user/apikeys", api.WithDBSession(api.mustAuth(api.userAPIKeyPOST)))
+	api.staticRouter.GET("/user/apikeys", api.WithDBSession(api.mustAuth(api.userAPIKeyGET)))
+	api.staticRouter.DELETE("/user/apikeys/:id", api.WithDBSession(api.mustAuth(api.userAPIKeyDELETE)))
 
 	// Endpoints for email communication with the user.
 	api.staticRouter.GET("/user/confirm", api.WithDBSession(api.noAuth(api.userConfirmGET))) // TODO POST
-	api.staticRouter.POST("/user/reconfirm", api.WithDBSession(api.withAuth(api.userReconfirmPOST)))
+	api.staticRouter.POST("/user/reconfirm", api.WithDBSession(api.mustAuth(api.userReconfirmPOST)))
 	api.staticRouter.POST("/user/recover/request", api.WithDBSession(api.noAuth(api.userRecoverRequestPOST)))
 	api.staticRouter.POST("/user/recover", api.WithDBSession(api.noAuth(api.userRecoverPOST)))
 
 	api.staticRouter.POST("/stripe/webhook", api.WithDBSession(api.noAuth(api.stripeWebhookPOST)))
 	api.staticRouter.GET("/stripe/prices", api.noAuth(api.stripePricesGET))
 
-	api.staticRouter.GET("/.well-known/jwks.json", api.noAuth(api.wellKnownJwksGET))
+	api.staticRouter.GET("/.well-known/jwks.json", api.noAuth(api.wellKnownJWKSGET))
 }
 
 // noAuth is a pass-through method used for decorating the request and
 // logging relevant data.
-func (api *API) noAuth(h httprouter.Handle) httprouter.Handle {
+func (api *API) noAuth(h HandlerWithUser) httprouter.Handle {
 	return func(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 		api.logRequest(req)
-		h(w, req, ps)
+		h(nil, w, req, ps)
 	}
 }
 
-// withAuth ensures that the user making the request has logged in.
-func (api *API) withAuth(h httprouter.Handle) httprouter.Handle {
+// mustAuth ensures that the user making the request has logged in.
+func (api *API) mustAuth(h HandlerWithUser) httprouter.Handle {
```

Comment:
I don't have a strong preference. For reference in skyd we use `RequirePassword`.

Maybe we can have `mustAuth`/`requireAuth` and `canAuth` in case we have a handler that might rely on authentication but doesn't really need it to function.

Reply:
I didn't find a single handler that could use `canAuth`. The one that handles both unauthenticated and authenticated requests doesn't want to rely on the middleware in order to be able to do clever things with caching, so it uses `noAuth`.

I also liked `noAuth`/`withAuth` better. It seems like Chris doesn't mind, so I'll revert back to that.